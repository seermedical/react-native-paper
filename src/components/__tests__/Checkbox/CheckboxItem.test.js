import * as React from 'react';
import { Platform } from 'react-native';
import renderer from 'react-test-renderer';
import { render } from 'react-native-testing-library';
import Checkbox from '../../Checkbox';

it('renders unchecked', () => {
  const tree = renderer
    .create(<Checkbox.Item status="unchecked" label="Unchecked Button" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render the iOS checkbox on different platforms', () => {
  Platform.OS = 'android';
  const tree = renderer
    .create(
      <Checkbox.Item status="unchecked" label="iOS Checkbox" mode="ios" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render the Android checkbox on different platforms', () => {
  Platform.OS = 'ios';
  const tree = renderer
    .create(
      <Checkbox.Item status="unchecked" label="iOS Checkbox" mode="android" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render leading checkbox control', () => {
  Platform.OS = 'ios';
  const tree = renderer
    .create(
      <Checkbox.Item
        label="Default with leading control"
        status={'unchecked'}
        mode="ios"
        position="leading"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should have `accessibilityState={ checked: true }` when `status="checked"`', () => {
  const { getByA11yState } = render(
    <Checkbox.Item status="checked" label="Checked Button" />
  );

  const element = getByA11yState({ checked: true });
  expect(element).toBeTruthy();
});

it('should have `accessibilityState={ checked: false }` when `status="unchecked"', () => {
  const { getByA11yState } = render(
    <Checkbox.Item status="unchecked" label="Unchecked Button" />
  );

  const element = getByA11yState({ checked: false });
  expect(element).toBeTruthy();
});

it('should have `accessibilityState={ checked: false }` when `status="indeterminate"', () => {
  const { getByA11yState } = render(
    <Checkbox.Item status="indeterminate" label="Indeterminate Button" />
  );

  const element = getByA11yState({ checked: false });
  expect(element).toBeTruthy();
});

it('disables the row when the prop disabled is true', () => {
  const { getByA11yLabel } = render(
    <Checkbox.Item accessibilityLabel="some checkbox" disabled />
  );

  const touchable = getByA11yLabel('some checkbox');

  expect(touchable.props).toMatchObject({
    accessibilityState: { disabled: true },
  });
});
