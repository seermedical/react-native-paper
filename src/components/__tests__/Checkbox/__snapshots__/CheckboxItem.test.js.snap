// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render leading checkbox control 1`] = `
<View
  accessibilityLabel="Default with leading control"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      undefined,
    ]
  }
>
  <View
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "padding": 6,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#03dac4",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 16,
            },
            Object {
              "color": "#000000",
              "textAlign": "right",
            },
            undefined,
          ],
        ]
      }
    >
      Default with leading control
    </Text>
  </View>
</View>
`;

exports[`can render the Android checkbox on different platforms 1`] = `
<View
  accessibilityLabel="iOS Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      undefined,
    ]
  }
>
  <View
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 16,
            },
            Object {
              "color": "#000000",
              "textAlign": "left",
            },
            undefined,
          ],
        ]
      }
    >
      iOS Checkbox
    </Text>
    <View
      accessibilityLiveRegion="polite"
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "height": 36,
            "padding": 6,
            "width": 36,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "borderColor": "rgba(0, 0, 0, 0.54)",
                "borderWidth": 0,
                "height": 14,
                "width": 14,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`can render the iOS checkbox on different platforms 1`] = `
<View
  accessibilityLabel="iOS Checkbox"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      undefined,
    ]
  }
>
  <View
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 16,
            },
            Object {
              "color": "#000000",
              "textAlign": "left",
            },
            undefined,
          ],
        ]
      }
    >
      iOS Checkbox
    </Text>
    <View
      accessibilityLiveRegion="polite"
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "padding": 6,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#03dac4",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders unchecked 1`] = `
<View
  accessibilityLabel="Unchecked Button"
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      undefined,
    ]
  }
>
  <View
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 16,
            },
            Object {
              "color": "#000000",
              "textAlign": "left",
            },
            undefined,
          ],
        ]
      }
    >
      Unchecked Button
    </Text>
    <View
      accessibilityLiveRegion="polite"
      accessibilityRole="checkbox"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "padding": 6,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#03dac4",
                "fontSize": 24,
              },
            ]
          }
        >
          □
        </Text>
      </View>
    </View>
  </View>
</View>
`;
